main.js

```diff
@@ -1,77 +1,57 @@
-//Преобразование формата даты
-function transformDate(date) {
-  const re = /(?<year>\d+)+-+(?<month>\d+)+-+(?<day>\d+)/gm;
-  const newDate = date
-    .replace(re, "$<day> $<month> $<year>")
-    .replace(/\s/g, ".");
-  return (newDate);
-}
-
-const date1 = "2020-11-26";
-console.log(transformDate(date1)); // 26.11.2020
-
-const date2 = "2020-1-5";
-console.log(transformDate(date2)); // 5.1.2020
+const obj1 = {
+    a: 'a',
+    b: {
+        a: 'a',
+        b: 'b',
+        c: {
+            a: 1,
+        },
+    },
+};
+const obj2 = {
+    b: {
+        c: {
+            a: 1,
+        },
+        b: 'b',
+        a: 'a',
+    },
+    a: 'a',
+};
+const obj3 = {
+    a: {
+        c: {
+            a: 'a',
+        },
+        b: 'b',
+        a: 'a',
+    },
+    b: 'b',
+};
 
-const date3 = "22-01-5";
-console.log(transformDate(date3)); // 5.01.22
+function deepEqual(a, b) {
+    if (a === b) {
+        return true;
+    }
 
-//Поиск объектов размещения:
-function searchData(word) {
-  const data = [
-    {
-      country: "Russia",
-      city: "Saint Petersburg",
-      hotel: "Hotel Leopold",
-    },
-    {
-      country: "Spain",
-      city: "Santa Cruz de Tenerife",
-      hotel: "Apartment Sunshine",
-    },
-    {
-      country: "Slowakia",
-      city: "Vysokie Tatry",
-      hotel: "Villa Kunerad",
-    },
-    {
-      country: "Germany",
-      city: "Berlin",
-      hotel: "Hostel Friendship",
-    },
-    {
-      country: "Indonesia",
-      city: "Bali",
-      hotel: "Ubud Bali Resort&SPA",
-    },
+    if (a == null || typeof(a) != "object" ||
+        b == null || typeof(b) != "object")
```

Comment:
This check raises questions.
The function should compare objects only according to the task.